test/e2e/specs/editor/various/block-comments.spec.js

```diff
@@ -283,44 +283,147 @@ test.describe( 'Block Comments', () => {
 	} );
 
 	test.describe( 'Keyboard navigation', () => {
-		test( 'should expand or collapse a comment with Enter key', async ( {
-			editor,
+		const KEY_COMBINATIONS = [
+			{
+				keyToExpand: 'Enter',
+				keyToCollapse: 'Enter',
+				keyName: 'enter',
+			},
+			{
+				keyToExpand: 'ArrowRight',
+				keyToCollapse: 'ArrowLeft',
+				keyName: 'arrow right and left',
+			},
+		];
+		KEY_COMBINATIONS.forEach(
+			( { keyToExpand, keyToCollapse, keyName } ) => {
+				test( `should expand or collapse a comment with ${ keyName } key`, async ( {
+					page,
+					editor,
+					blockCommentUtils,
+				} ) => {
+					await blockCommentUtils.addBlockWithComment( {
+						type: 'core/heading',
+						attributes: { content: 'Testing block comments' },
+						comment: 'Test comment',
+					} );
+
+					// Click on the title field to deselect the block and the comment.
+					await editor.canvas
+						.getByRole( 'textbox', { name: 'Add title' } )
+						.focus();
+
+					const thread = page
+						.getByRole( 'region', {
+							name: 'Editor settings',
+						} )
+						.getByRole( 'treeitem', {
+							name: 'Note: Test comment',
+						} );
+
+					// Expand the comment with the specified key.
+					await thread.focus();
+					await page.keyboard.press( keyToExpand );
+					await expect(
+						thread,
+						'comment should be expanded with $keyToExpand key'
+					).toHaveAttribute( 'aria-expanded', 'true' );
+
+					// The related block should be selected, but the focus should remain on the comment.
+					await expect(
+						editor.canvas.getByText( 'Testing block comments' )
+					).toHaveClass( /is-selected/ );
+					await expect( thread ).toBeFocused();
+
+					// Collapse the comment with the specified key.
+					await page.keyboard.press( keyToCollapse );
+					await expect(
+						thread,
+						'comment should be collapsed with $keyToCollapse key'
+					).toHaveAttribute( 'aria-expanded', 'false' );
+				} );
+			}
+		);
+
+		test( 'should move to the adjacent comment with arrow keys', async ( {
```

Comment:
The diffs are not showing well, but I simply added the following two new tests.

- `should move to the adjacent comment with arrow keys`
- `should move to the first or last comment with Home or End keys`